mlrun/api/utils/clients/iguazio.py

```diff
@@ -0,0 +1,78 @@
+import copy
+import typing
+
+import requests.adapters
+import urllib3
+
+import mlrun.api.schemas
+import mlrun.api.utils.projects.remotes.member
+import mlrun.errors
+import mlrun.utils.singleton
+from mlrun.utils import logger
+
+
+class Client(metaclass=mlrun.utils.singleton.Singleton,):
+    def __init__(self) -> None:
+        super().__init__()
+        http_adapter = requests.adapters.HTTPAdapter(
+            max_retries=urllib3.util.retry.Retry(total=3, backoff_factor=1)
+        )
+        self._session = requests.Session()
+        self._session.mount("http://", http_adapter)
+        self._api_url = mlrun.mlconf.iguazio_api_url
+
+    def get_grafana_service_url_if_exists(
+        self, session_cookie: str
+    ) -> typing.Optional[str]:
+        logger.debug("Getting grafana service url from Zebo")
+        response = self._send_request_to_api(
+            "GET", "app_services_manifests", session_cookie
+        )
+        response_body = response.json()
+        if len(response_body.get("data", [{}])) > 1:
```

Comment:
We will support several sets at some point - suggest to build for it already - just iterate over them and return the first grafana from the first set you find it on